pkg/cmd/pulumi-language-java/main.go

```diff
@@ -174,8 +174,19 @@ func (host *javaLanguageHost) determinePulumiPackages(
 
 	// Run our classpath introspection from the SDK and parse the resulting JSON
 	cmd := host.exec.cmd
-	args := host.exec.pluginArgs
-	output, err := host.runJavaCommand(ctx, cmd, args)
+
+	// Build the project, if possible, to avoid user errors during bootstrap
```

Comment:
I'm concerned this is slightly slowing every operation down as it's more subprocess interaction; have we considered instead downgrading the error to a silent info-level message, say `output, err := host.runJavaCommand(ctx, cmd, pluginArgs)` - ignore the err, return `[]plugin.PulumiPluginJSON{}`. Plugin determination is an advisory feature so it doe not need to escalate to an error.

Reply:
Not a fan of hiding error messages, but I don't have a better idea either ;)
So I've deleted the build step from this PR and tried to address this concern in a separate PR:  https://github.com/pulumi/pulumi-java/pull/584